r1211266 | lbeltrame | 2011-01-03 14:41:10 +0100 (Mon, 03 Jan 2011) | 2 lines
Changed paths:
   M /trunk/KDE/kdebindings/python/pykde4/sip/kdecore/typedefs.sip

Fix building against PyQt 4.8.2 (forwardport of r1208598 by sedwards)

Applied-Upstream: 4.6.0

Index: kdebindings/python/pykde4/sip/kdecore/typedefs.sip
===================================================================
--- kdebindings/python/pykde4/sip/kdecore/typedefs.sip	(revision 1211265)
+++ kdebindings/python/pykde4/sip/kdecore/typedefs.sip	(revision 1211266)
@@ -582,90 +582,6 @@
 %End
 };
 
-template <TYPE>
-%MappedType QSet<TYPE>
-{
-%TypeHeaderCode
-#include <qset.h>
-%End
-
-%ConvertFromTypeCode
-    // Create the list.
-    PyObject *l;
-
-    if ((l = PyList_New(sipCpp->size())) == NULL)
-        return NULL;
-
-    // Set the list elements.
-    QSet<TYPE> set = *sipCpp;
-    int i = 0;
-    foreach (TYPE value, set)
-    {
-        PyObject *obj = sipConvertFromNewType(&value, sipType_TYPE, sipTransferObj);
-        if (obj == NULL || PyList_SET_ITEM (l, i, obj) < 0)
-        {
-            Py_DECREF(l);
-
-            if (obj)
-                Py_DECREF(obj);
-
-            return NULL;
-        }
-
-        Py_DECREF(obj);
-        i++;
-    }
-   
-    return l;
-%End
-
-%ConvertToTypeCode
-    // Check the type if that is all that is required.
-    if (sipIsErr == NULL)
-    {
-        if (!PyList_Check(sipPy))
-            return 0;
-    }
-
-    // Check the type if that is all that is required.
-    if (sipIsErr == NULL)
-    {
-        if (!PyList_Check(sipPy))
-            return 0;
-
-        for (int i = 0; i < PyList_GET_SIZE(sipPy); ++i)
-            if (!sipCanConvertToType(PyList_GET_ITEM (sipPy, i), sipType_TYPE, SIP_NOT_NONE))
-                return 0;
-    }
-
-    QSet<TYPE> *qs = new QSet<TYPE>;
- 
-    for (int i = 0; i < PyList_GET_SIZE(sipPy); ++i)
-    {
-        int state;
-
-        TYPE *t = reinterpret_cast<TYPE *>(sipConvertToType(PyList_GET_ITEM (sipPy, i), sipType_TYPE, sipTransferObj, SIP_NOT_NONE, &state, sipIsErr));
- 
-        if (*sipIsErr)
-        {
-            sipReleaseType(t, sipType_TYPE, state);
-
-            delete qs;
-            return 0;
-        }
-
-        *qs << *t;
-
-        sipReleaseType(t, sipType_TYPE, state);
-    }
- 
-    *sipCppPtr = qs;
- 
-    return sipGetState(sipTransferObj);
-%End
-};
-
-
 template <TYPE1,TYPE2>
 %MappedType QPair<TYPE1,TYPE2>
 {
